### PR TITLE
implement Upload.terminate

### PR DESCRIPTION
**Pull request by ifedapoolarewaju**

### CODE REVIEW

File: lib/upload.js [hunk 1]
```diff
@@ -242,6 +272,16 @@ class Upload {
       clearTimeout(this._retryTimeout);
       this._retryTimeout = null;
     }
+
+    if (shouldTerminate) {
+      Upload.terminate(this.url, this.options, (err, xhr) => {
+        if (err || !this._hasStorage()) {
```
Comment (Acconut) [inline lib/upload.js]: Can you explain what the purpose of `|| !this._hasStorage()` here is?

File: lib/upload.js [hunk 2]
```diff
@@ -242,6 +272,16 @@ class Upload {
...
+          return cb(err, xhr);
+        }
+
+        this._storage.removeItem(this._fingerprint, cb);
+      });
+    }
```
Comment (Acconut) [inline lib/upload.js]: The callback should also be invoked if `shouldTerminate === false`, just to have a consistent API interface.

File: test/spec/upload.js [hunk 3]
```diff
@@ -1025,7 +1025,22 @@ describe("tus", function () {
...
+          }
+
+          // delete the upload after it was completed
+          upload.abort(true, function (err) {
+            if (err) {
+              done.fail(err);
+              return;
+            }
+
+            validateUploadDeletion(upload, done);
```
Comment (Acconut) [inline test/spec/upload.js]: Can we have a test that verifies that the terminate method sends a DELETE request to the provided upload URL?

File: lib/upload.js [hunk 4]
```diff
@@ -74,6 +74,36 @@ class Upload {
     this._offsetBeforeRetry = 0;
   }
 
+  static terminate(url, options, cb) {
```
Comment (Acconut) [inline lib/upload.js]: Please add documentation to the README for the two new public methods.

File: test/spec/upload.js (hunk 3 above)
Comment (ifedapoolarewaju) [reply test/spec/upload.js]: added

File: lib/upload.js (hunk 4 above)
Comment (ifedapoolarewaju) [reply lib/upload.js]: added

File: lib/upload.js (hunk 2 above)
Comment (ifedapoolarewaju) [reply lib/upload.js]: fixed

File: lib/upload.js (hunk 1 above)
Comment (ifedapoolarewaju) [reply lib/upload.js]: I have refactored the code so the intention is clearer

**Pull request merged**